### ISSUE TITLE
Provide aria-expanded attribute to calcite-button

### ISSUE THREAD

**Issue by shellchulx** [Esri/calcite-design-system#14404]
**monday.com sync:** #11972158045

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Dropdown

### Summary

https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/67129
https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/67142
https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/67149
https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/67160

The aria attributes are not added to the internal trigger button element. 

We need to ensure custom controls provide proper textual name, role, and state information - ENT-AMP315
cc: @catherinehynes @mpriour 

### Actual Behavior

Screen reader users will be unable to determine that these controls expand and collapse content. They will also have difficulty determining whether a control is currently expanded or collapsed.

### Expected Behavior

The trigger button doesn't have the ability to display its state - expanded or collapsed

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/dropdown/ 
Please inspect this `Select landform` button in the console window when the menu is closed or opened.

### Reproduction Steps
Please refer to the Reproduction Sample

### Reproduction Version

The current version

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise, ArcGIS Online

**Comment by geospatialem** [comment]
Added the above to the larger epic for consistency across components, similar to the offering of an `aria` property to `calcite-action`.

**Comment by geospatialem** [comment]
cc @Anastasiia-Boleiko

**Comment by mpriour** [comment]
This is partially fixed by using the new `referenceElement` pattern and **not** using the now deprecated `trigger` slot. The specific `aria-expanded` property is fully supported with the reference element pattern and the aria attribute goes on the internal native button element correctly. However, the lack of correct `aria-controls` and `aria-haspopover` attributes on the reference element trigger prevent recognition of the dropdown as a menu by screen readers. I have specifically raised that issue in #14644

**Comment by paulcpederson** [comment]
I'm also now getting issues from an accessibility audit about this issue. Can the calcite team look at adding the has popup and other aria attributes to the button for the dropdown so that screen readers properly understand this as a menu?

Looking through the accessibility requirements I think there are only a couple that would need added:

> - The element with role button has "aria-haspopup" set to either "menu" or "true".
> - When the menu is displayed, the element with role button has "aria-expanded" set to "true". When the menu is hidden, "aria-expanded" is set to "false".

@DitwanP @geospatialem I'm using a brand new NVDA on firefox if that's important. IMO we should up-level this as dropdown has heavy usage and many products will be getting flagged for accessibility violations until it's fixed.
